testsuite/tests/lib-bigarray/pr11118.ml

```diff
@@ -0,0 +1,16 @@
+(* TEST
+*)
+
+(* PR#11118 set data in polymorphic bigarray with offset > INT32_MAX *)
+
+let set_value ba i x =
+  Bigarray.Array1.set ba i x
+
+let () =
+  let max_int32 = 1 lsl 31 in
+  let len = max_int32 + 1 in
+  let bigarray = Bigarray.Array1.create Bigarray.Char Bigarray.c_layout len in
```

Comment:
This allocates a 2G bigarray, not sure if this can cause problems in certain architectures...